panther_battery/src/battery_publisher.cpp

```diff
@@ -43,6 +48,10 @@ void BatteryPublisher::Publish()
   } catch (const std::runtime_error & e) {
     RCLCPP_ERROR_THROTTLE(
       node_->get_logger(), *node_->get_clock(), 1000, "Error reading battery data: %s. ", e.what());
+
+    diagnostic_updater_->broadcast(
+      diagnostic_msgs::msg::DiagnosticStatus::ERROR,
+      "Error reading battery data: " + std::string(e.what()));
```

Comment:
The reason is the `broadcast` method. The hardware ID is available only for diagnostic tasks. 

Possible options are:
- fix this using aggregation (possibly in the future)
- use a raw publisher
- try to implement diagnostic task (cumbersome, since we would probably need dig into `Battery` implementation)

Reply:
This explanation satisfies me, no point in fixing this at this point but aggregation seems the best option